annotation-utils/src/main/java/edu/isi/vista/annotationutils/RestoreJson.kt

```diff
@@ -1,109 +1,149 @@
+                        val docID = Symbol.from(aceDocID)
+                        aceTextSource.getOriginalText(docID).orNull()
+                                ?: throw RuntimeException("Could not get original text for $docID")
+                    } else {
+                        // If it is a Gigaword doc: First 21 characters of the filename are the
+                        // document id
+                        // Example filename is AFP_ENG_19960918.0012-admin.json
+                        val docID = Symbol.from(filename.substring(0, 21))
+                        gigaWordTextSource.getOriginalText(docID).orNull()
+                                ?: throw RuntimeException("Could not get original text for $docID")
+                    }
+                    jsonTree.replaceFieldEverywhere("sofaString", text)
+                    val outFile = File(projectOutDir.toString(), filename)
+                    outFile.writeBytes(prettyPrinter.writeValueAsBytes(jsonTree))
+                    logger.info { "Restored $filename" }
+                } else {
+                    logger.warn { "Cannot restore $filename, only English is supported" }
+                }
+            }
+        }
+    }
+}
+
+fun makeTextSource(params: Parameters, corpusName: String): OriginalTextSource {
+    // Create an OriginalTextSource for getting original document text to put in json:
+    if(corpusName.equals("gigaword", ignoreCase = true)) {
+        // Create the gigaword corpus TextSource
+        val textMap = DocIDToFileMappings.forFunction { symbol ->
+            Optional.of(File(params.getExistingDirectory("gigawordDataDirectory"), docIDToFilename(symbol)))
+        }
+        val indexMap = DocIDToFileMappings.forFunction { symbol ->
+            Optional.of(File(params.getExistingDirectory("indexDirectory"), docIDToFilename(symbol) + ".index"))
+        }
+        return OffsetIndexedCorpus.fromTextAndOffsetFiles(textMap, indexMap)
+
+    } else if (corpusName.equals("ace", ignoreCase = true)) {
+        // Create the Ace corpus TextSource
+        return AceCorpusTextSource(params.getExistingDirectory("aceEngDataDirectory"))
+    } else {
+        throw IOException("A corpus: " + corpusName + " does not exist.");
+    }
+}
+
+fun docIDToFilename(symbol: Symbol?): String {
+    if (symbol == null) {
+        throw RuntimeException("Got null symbol")
+    }
+    // e.g. AFP_ENG_19960918.0012 -> afp_eng/afp_eng_199609
+    val st = symbol?.asString()?.toLowerCase()
+    val dir = st?.substring(0, 7) //
+    val basename = st?.substring(0, 14)
+    return Paths.get(dir, basename).toString()
+}
+
+private fun Path.removePrefixPath(prefix: Path): Path {
+    require(this.startsWith(prefix)) {
+        "The path $this does not start with the prefix $prefix"
+    }
+    return this.subpath(prefix.nameCount, this.nameCount)
+}
+
+
+private fun getAceDocID(filename: String): String? {

```

Comment:
This function returns the event.subtype as a string if the document is an ACE document, else it returns null.
After some discussion with Liz, this regex seemed the a good option to differentiate ace and giga-word documents and extract the event detail at the same time. (Although currently, ace documents have `xmi`, in their name, using this to differentiate would not be future-proof)